In [1]:
import pandas as pd
import numpy as np
from scipy import stats

# Đọc file
df = pd.read_csv("../data_2025/sample/sample_100.csv", dtype={"SBD": str})


In [2]:
from scipy import stats
#Tinh zscore
df["zscore_math"] = stats.zscore(df["TOAN"])
df
# df[(abs(df['zscore_math']) > 3)]

SBD  TOAN  zscore_math
0   02083823  5.00     0.025282
1   38012819  4.00    -0.555909
2   34006288  8.00     1.768854
3   08000914  5.25     0.170580
4   01015076  8.00     1.768854
..       ...   ...          ...
95  02081005  6.75     1.042366
96  53006884  5.50     0.315877
97  37017504  6.50     0.897068
98  43002113  4.35    -0.352492
99  31007394  8.00     1.768854

[100 rows x 3 columns]

In [3]:
scores = df["TOAN"]

n = len(scores)
mean = np.mean(scores)
std = np.std(scores, ddof=1)  # độ lệch chuẩn mẫu

# Mức tin cậy 95%
alpha = 0.05
t_value = stats.norm.ppf(1 - alpha/2)

# Sai số chuẩn
se = std / np.sqrt(n)

# Khoảng tin cậy
ci_lower = mean - t_value * se
ci_upper = mean + t_value * se

print("Số lượng mẫu:", n)
print("Trung bình mẫu:", mean)
print("Độ lệch chuẩn mẫu:", std)
print(f"Khoảng tin cậy 95% cho kỳ vọng: [{ci_lower:.3f}, {ci_upper:.3f}]")

Số lượng mẫu: 100
Trung bình mẫu: 4.956500000000001
Độ lệch chuẩn mẫu: 1.7292734359151003
Khoảng tin cậy 95% cho kỳ vọng: [4.618, 5.295]


---
### Kiểm định giả thuyết
#### Theo kết quả phân tích điểm thi tốt nghiệp THPT năm 2024 của Bộ Giáo dục và Đào tạo, điểm trung bình môn toán của thí sinh cả nước là 6.45 điểm. Tuy nhiên, nhóm nghi ngờ rằng con số này có cải thiện đáng kể so với trước đây. Bởi vậy, nhóm tiến hành khảo sát 100 thí sinh và tính được trung bình mẫu là 4.96. Nhóm sẽ tiến hành các thí nghiệm để kết luận về nghi ngờ nói trên, với mức ý nghĩa 5%

- H0 = 6.45
- H1 = mean < 6.45

In [4]:
# Trường hợp chưa biết phương sai, cỡ mẫu n ≥ 30
H0 = 6.45
H1 = mean < 6.45
alpha = 0.05

# Miền bác bỏ (sử dụng phân phối chuẩn tắc)
z_alpha = stats.norm.ppf(1 - alpha)
print("Giá trị z_alpha (phân phối chuẩn tắc):", z_alpha)
print("Miền bác bỏ: (-∞,", -z_alpha, ")")

# Phương sai hiệu chỉnh
s_squared = 1/(n-1) * sum((scores - mean)**2)
s = np.sqrt(s_squared)
print("Phương sai hiệu chỉnh:", s_squared)

# Giá trị quan sát
z_qs = (mean - H0) / s * np.sqrt(n)
print("Giá trị z_qs:", z_qs)
if z_qs < -z_alpha:
    print(">>> Bác bỏ giả thuyết H0")
else:
    print(">>> Chưa có cơ sở để bác bỏ giả thuyết H0")

Giá trị z_alpha (phân phối chuẩn tắc): 1.6448536269514722
Miền bác bỏ: (-∞, -1.6448536269514722 )
Phương sai hiệu chỉnh: 2.9903866161616173
Giá trị z_qs: -8.636575159148649
>>> Bác bỏ giả thuyết H0
